### ISSUE TITLE
Use a static site generator

### ISSUE THREAD

**Issue by kleinpa** [dspzl/dspzl.org#19]

**Comment by schwalbs** [comment]
> Whatever you pick, it'll seem like a weird choice in a couple years.

Fair enough, I guess I don't care what library we use, but more the language. At least if he wants reasonable feedback from us and to help future webmasters. Python and Node will give more applicable experience to the actives and I just don't feel Ruby is a commonly used.

Plus like I said earlier, I won't be of much use on those PRs and I'd like to help out when possible.

@DavidPurdy1 ultimately up to you and I'll support whatever you go with, just throwing my 2 cents in here.

**Comment by yorker78** [comment]
I agree with the sentiment "use whatever you want" as you have to deal with it way more than we do, so consider your vote to be the "majority" no matter what we say. 

I didn't really like Jekyll (or any static site generator) at the time which is why I never merged that PR. I was of the opinion that, while it is easier once you know what you're doing with them, it is one more hurdle of learning before changes can be made. It's easy to look back and say "yeah, I knew all of that stuff in college", but I didn't think it was a good idea to add a dependency for people still learning especially when you possibly won't even be physically in the same room for development to help them along the way. I remember Gartzke re-did the entire site because it was using a static site generator and he switched everything to a more pure HTML/CSS/JS with a couple css/js libraries like materialize. Was the code good? It was and has been serviceable. But more importantly it was maintainable by anyone who took the 200 level Web Dev class at Rose. 

Granted you should take what I say with a grain of salt as I'm behind the times when it comes to front end because it seems like no matter what industry I'm in, I'm just going to be fixing legacy Java code for the rest of my life. Thankfully I'm no longer doing full stack, so I'm probably even less useful as a resource now, but as a boomer alumni I am always ready to give my opinion, solicited or otherwise.

**Comment by yorker78** [comment]
Not to add to my already long comment, but there is one difference and advantage the current and future webmasters have, which is this repo. We made it while I was webmaster because there was a whole thing about Rose not hosting for us and we had to come up with a new solution. So there is now a bit more of a "webmaster community" to reach out to for help. So that is also something to consider, maybe my experience and how I planned for the future is no longer relevant as we aren't just passing around a zip file with the code and FTP credentials to update the site.

**Comment by DavidPurdy1** [comment]
Sorry, I was out camping for the weekend. Thanks for the suggestions. I think I might go with eleventy, I looked into a couple and that one lets you use whatever with node, which we learn in the web intro class so brothers should probably be familiar with it. I would like a static site generator so if the webmaster doesn't really care too much they can just update the site from a data file or whatever. I will start out with just getting a branch in with the stuff from the current website tonight and then go from there. If it starts getting too complicated I might just ditch it.

**Comment by DavidPurdy1** [comment]
Alright so I looked at a static site generator about a week ago. After setting it up, I kinda realized it wasn't much easier imo. I have a couple things I could probably do to make the website easier to update, but right now, I don't think adding a site generator will help much.

**Comment by kleinpa** [comment]
Fair enough. The Web Components trick being used now does solve many of the same problems a static site generator would.
